packages/verification-client/src/Client.js

```diff
@@ -96,6 +123,191 @@ export default class Client {
     })
   }
 
+  /**
+   * Verifies a file hash based on claims
+   * @param {string} hash
+   * @return {FileVerification}
+   */
+  async verifyFileClaimBased (hash) {
+    let result=await this.resolveAndValidateFileClaim(hash)
+    return result
+  }
+
+  async resolveAndValidateFileClaim(hash) {
+    // Get Events for Filehash
+    const fileEvents = await this.claimContract.getPastEvents(
+        'Claim', {
+          filter: {file: hash},
+          fromBlock: 0,
+        })
+
+    let registered, revoked = false;
+    let issuerAddr, issuerName;
+    let issuerVerified;
+    let expiry, issuerImg;
+
+
+    console.log(fileEvents.length+" event(s) found on Blockchain for File "+hash)
+    // For Each Event
+    for (const fileEvent of fileEvents) {
+
+
+      console.log("---------")
+      console.log("Processing event #"+(i+1))
+
+      //Get Claim Hash from Event
+      let fileHash = fileEvent.returnValues.file
+      let claimHash = fileEvent.returnValues.hash
+      if (fileHash == hash){
+        console.log("Event is for File.")
+        console.log(fileEvent)
+        console.log("Etherscan link to Tx: https://ropsten.etherscan.io/tx/"+fileEvent.transactionHash)
+        console.log("File is associated with Claim Hash: "+claimHash)
+
+
+        //Get Claim from endpoint for Claimhash
+        let claim;
+        try {
+          claim = await this.getRawClaim(claimHash)
+        } catch (e) {
+          console.error("Could not retrieve Claim by Hash, discarding.", e)
+          continue;
+        }
+        let claimString = JSON.stringify(claim)
+        console.log("Raw JSON Claim: "+claimString)
+
+        //Verify Claim (hashes to claimhash, belongs to file, Has right content/type)
+        if (claim["@id"] === "cert:hash:"+fileHash) {
+          console.log("Hashes matching, Claim is for this File.")
+
+          //Get Issuer Hash from Address from Signature
+          issuerAddr=await this.verifySignatureAndGetPubkey(claim, claim.signature)
+          console.log("Recovered Signer address:", issuerAddr)
+
+          if (issuerAddr == claim.proof.creator){
```

Comment:
I'd inverse it and go for ``if (issuerAddr != claim.proof.creator) and then do the `continue`, no else needed:

```
if (issuerAddr != claim.proof.creator){
           console.error("Signer Address does NOT match Claim Creator attribute, discarding.")
            issuerAddr=null
            continue;
}

 console.error("Signer Address does NOT match Claim Creator attribute, discarding.")
  issuerAddr=null

```

Reply:
Thanks, addressed in f72b10b7e777f35c4fee40319d92682a0d190b0b